Lib/http/server.py

```diff
@@ -902,6 +937,24 @@ def guess_type(self, path):
             return guess
         return 'application/octet-stream'
 
+    def get_range(self):
+        """Return a tuple of (start, end) representing the range header in
+        the HTTP request. If the range header is missing or not resolvable,
+        None is returned. This only supports single part ranges.
+
+        """
+        range_header = self.headers.get('range')
+        if not range_header:
+            return None
+        m = re.match(r'bytes=(\d+)-(\d*)$', range_header)
```

Comment:
The header *might* also include the instance length unless it is unknown so, technically speaking, the header could be something like:

```text
HTTP/1.1 206 Partial Content
Date: Wed, 15 Nov 1995 06:25:24 GMT
Last-Modified: Wed, 15 Nov 1995 04:58:08 GMT
Content-Range: bytes 21010-47021/47022
Content-Length: 26012
Content-Type: image/gif
```

Reply:
I think instance length will only be included in the header of the HTTP response, the example you gave is also an HTTP response. In L778 of server.py I've returned `fs.st_size` as instance length.

The client should never send the `Range` header with the total length, this part of the code is used to parse the `Range` header in HTTP *request* sent by the client, so there should be no need to deal with instance length?

(I apologize for any misunderstanding that may have been caused by the name `get_range`, I'll change it to `parse_range`.)

Reply:
Oups, you're right! Sorry for the false alarm.